test/testscript.go

```diff
@@ -0,0 +1,155 @@
+package test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"text/template"
+
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+type ProjectTest struct {
+	t          *testing.T
+	params     testscript.Params
+	setupFuncs []func(*testscript.Env) error
+	projectDir string
+}
+
+func NewProjectTest(t *testing.T, projectDir string) *ProjectTest {
+	params := testscript.Params{
+		Files:               []string{},
+		Setup:               nil,
+		RequireExplicitExec: true,
+		RequireUniqueNames:  true,
+	}
+	return &ProjectTest{
+		t:          t,
+		params:     params,
+		projectDir: projectDir,
+	}
+}
+
+func (e *ProjectTest) CopyFile(srcRelPath string) *ProjectTest {
+	setupFunc := func(env *testscript.Env) error {
+		src := filepath.Join(e.projectDir, srcRelPath)
+		dst := filepath.Join(env.WorkDir, srcRelPath)
+		dstDir := filepath.Dir(dst)
+		if err := os.MkdirAll(dstDir, 0755); err != nil {
+			msg := "failed to create destination directory %s: %w"
+			return fmt.Errorf(msg, dstDir, err)
+		}
+
+		srcContent, err := os.ReadFile(src)
+		if err != nil {
+			return fmt.Errorf("failed to read source file %s: %w", src, err)
+		}
+
+		if err := os.WriteFile(dst, srcContent, 0644); err != nil {
+			msg := "failed to write destination file %s: %w"
+			return fmt.Errorf(msg, dst, err)
+		}
+		return nil
+	}
+	e.setupFuncs = append(e.setupFuncs, setupFunc)
+	return e
+}
+
+func (e *ProjectTest) RenderTemplateFileFromEnvWithCleanup(
+	srcRelPath string,
+	envKeys []string,
+) *ProjectTest {
+	src := filepath.Join(e.projectDir, srcRelPath)
+	tmpl := template.Must(template.ParseFiles(src))
+	dstRelPath := srcRelPath
+	return e.RenderTemplateFromEnvWithCleanup(tmpl, dstRelPath, envKeys)
+}
+
+func (e *ProjectTest) RenderTemplateStringFromEnvWithCleanup(
+	templateString string,
+	dstRelPath string,
+	envKeys []string,
+) *ProjectTest {
+	tmpl := template.Must(
+		template.New(filepath.Base(dstRelPath)).
+			Parse(templateString),
+	)
+	return e.RenderTemplateFromEnvWithCleanup(tmpl, dstRelPath, envKeys)
+}
```

Comment:
these Render wrappers allow for the generation of go templates from either a file (such as `fn-call.json` in the coingecko example) or from a string (such as the config template in `test/live/live_test.go`).